### ISSUE TITLE
Issues with GlobalVariablesSniff

### ISSUE THREAD

**Issue by jrfnl** [WordPress-Coding-Standards/WordPress-Coding-Standards#300]
Ok, so I ran into a lot of warnings with the GlobalVariablesSniff where local function variables where flagged because they had the same name as a global variable. While this may not be good practice/could be confusing, the sniff is wrongly flagging these uses as errors.

Additionally complex variables are not being flagged while they should be.

So I created a number of extra unit tests. Haven't figured out how to fix the sniff yet, but I thought I'd put the tests out there for whomever wants to have a go at this. Should I send in a PR for this ? The build obviously wouldn't pass as is.

The extra tests can be found here: https://github.com/jrfnl/WordPress-Coding-Standards/commit/54a9352db45c6ef40c195a30aec1ad292c3dd8f4

**Comment by westonruter** [comment]
Thanks for the link to the commit. When someone wants to pick this up, they can cherry-pick your commit into a new feature branch and then issue a PR when the additional changes have been made.

**Comment by lkraav** [comment]
How's this doing? On 0.6.0, I'm getting flagged for stuff like

```php
add_filter( "comments_open", function( $open, $post_id ) {

    $post = get_post( $post_id );

    return "post" === $post->post_type;

}, 10, 2 );
```

**Comment by westonruter** [comment]
The sniff needs to be smarter about determining whether the `$post` is being assigned in the global scope, or if it is inside of a function and `global $post;` is present. In other cases, it should not be flagged as an issue.

**Comment by westonruter** [comment]
Duplicate: #486

**Comment by westonruter** [comment]
- [ ] Also exclude global variable overrides from being reported as errors if the instance occurs inside of a class method that starts with `test_`, and thus is interpreted to be part of a unit test (which has to override globals often).
